File: app/src/main/java/com/riggle/plug/utils/event/SimpleApiResponse.java
package com.riggle.plug.utils.event;

import com.google.gson.annotations.SerializedName;
import com.riggle.plug.data.api.ApiResponse;

public class SimpleApiResponse<T> {

    @SerializedName("error")
    protected ErrorResponse error;
    @SerializedName("status")
    protected boolean status;
    @SerializedName("success")
    protected ApiResponse<T> success;

    public ErrorResponse getError() {
        return error;
    }

    public void setError(ErrorResponse error) {
        this.error = error;
    }

    public boolean isStatus() {
        return status;
    }

    public void setStatus(boolean status) {
        this.status = status;
    }

    public ApiResponse<T> getSuccess() {
        return success;
    }

    public void setSuccess(ApiResponse<T> success) {
        this.success = success;
    }

}

File: app/src/main/java/com/riggle/finza_finza/data/api/SimpleApiResponse.kt
package com.riggle.finza_finza.data.api

import com.google.gson.annotations.SerializedName
import java.io.Serializable

open class SimpleApiResponse : Serializable {

    @SerializedName("statusCode")
    var statusCode: Int? = null
        protected set

    @SerializedName("message")
    var message: String? = null
        protected set

    override fun toString(): String {
        return "SimpleApiResponse{" +
                "success=" + statusCode +
                ", message='" + message + '\'' +
                '}'
    }
}